h2o-algos/src/main/java/hex/gam/GAM.java

```diff
@@ -310,115 +389,259 @@ protected int nModelsInParallel(int folds) {
   }
 
   private class GAMDriver extends Driver {
-    double[][][] _zTranspose; // store for each GAM predictor transpose(Z) matrix
-    double[][][] _penalty_mat_center;  // store for each GAM predictor the penalty matrix
-    double[][][] _penalty_mat;  // penalty matrix before centering
-    public double[][][] _binvD; // store BinvD for each gam column specified for scoring
-    public int[] _numKnots;  // store number of knots per gam column
-    String[][] _gamColNames;  // store column names of GAM columns
-    String[][] _gamColNamesCenter;  // gamColNames after de-centering is performed.
+    double[][][] _zTranspose;         // store transpose(Z) matrices for CS and TP smoothers
+    double[][][] _zTransposeCS;       // store transpose(zCS) for thin plate smoother to remove optimization constraint
+    double[][][] _penalty_mat_center; // store centered penalty matrices of all smoothers
```

Comment:
Is there some reason for using an underscore instead of camel case?

Reply:
I changed it to camel case.